lib/jxl/dec_modular.cc

```diff
@@ -474,6 +475,7 @@ Status ModularFrameDecoder::ModularImageToDecodedRect(
       } else if (rgb_from_gray) {
         c_in = 0;
       }
+      JXL_ASSERT(c_in < gi.channel.size());
```

Comment:
please add a similar check for `c < gi.channel.size()` near line 554

Reply:
Done